crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -375,12 +379,20 @@ fn simplify_call(func: ValueId, arguments: &[ValueId], dfg: &mut DataFlowGraph)
         Value::Intrinsic(intrinsic) => *intrinsic,
         _ => return None,
     };
-    let constant_args: Option<Vec<_>> =
-        arguments.iter().map(|value_id| dfg.get_numeric_constant(*value_id)).collect();
-    let constant_args = match constant_args {
-        Some(constant_args) => constant_args,
-        Option::None => return None,
+
+    // Fetch args for intrinsic functions with only constant args
+    let constant_args = match intrinsic {
+        Intrinsic::ToBits(_) | Intrinsic::ToRadix(_) => {
+            let constant_args: Option<Vec<_>> =
+                arguments.iter().map(|value_id| dfg.get_numeric_constant(*value_id)).collect();
+            match constant_args {
+                Some(constant_args) => constant_args,
+                Option::None => return None,
```

Comment:
Instead of checking the intrinsic type here and returning early if we don't have all constant args, I think we should leave it as an Option and only have an `if let Some(args) = constant_args {` check in the intrinsics that need it below line 397.

Reply:
Switched